Daniel Choe/Week_01/Homework_03/js/exercises.js

```diff
@@ -0,0 +1,263 @@
+
+
+// # Credit Card Validation
+//
+// You're starting your own credit card business. You've come up with a new way
+//to validate credit cards with a simple function called validateCreditCard that returns true or false.
+//
+// Here are the rules for a valid number:
+//
+// - Number must be 16 digits, all of them must be numbers
+// - You must have at least two different digits represented (all of the digits cannot be the same)
+// - The final digit must be even
+// - The sum of all the digits must be greater than 16
+//
+// The following credit card numbers are valid:
+//
+// - `9999-9999-8888-0000`
+// - `6666-6666-6666-1666`
+//
+// The following credit card numbers are invalid:
+//
+// - `a923-3211-9c01-1112` invalid characters
+// - `4444-4444-4444-4444` only one type of number
+// - `1111-1111-1111-1110` sum less than 16
+// - `6666-6666-6666-6661` odd final number
+//
+// ## Example
+// ```
+// validateCreditCard('9999-9999-8888-0000'); // Returns: true
+// ```
+//
+// *Hint*: Remove the dashed from the input string before checking if the input credit card number is valid.
+//
+// *Bonus*: Return an object indicating whether the credit card is valid, and if not, what the error is
+//
+// ```
+// { valid: true, number: 'a923-3211-9c01-1112' }
+// { valid: false, number: 'a923-3211-9c01-1112', error: ‘wrong_length’ }
+// ```
+//
+// *Double Bonus*: Make your credit card scheme even more advanced! What are the rules, and what are some numbers that pass or fail? Ideas: check expiration date! Check out the Luhn Algorithm for inspiration.
+//
+//
+
+creditCard1 = '1134-5678-9098-76544'; //17 digits should return false.
+creditCard2 = '1134-5678-9098-765a';  //has letters should return false.
+creditCard3 = '1134-5678-9098-7655';  //last digit not divisible by 2 should return false.
+creditCard4 = '1111-1111-1111-1111';  //last digit not divisible by 2 should return false.
+creditCard5 = '1111-1111-0000-1112';  //sum is less than 16 should return false.
+creditCard6 = '9999-9999-8888-0002';  //should return true.
+creditCard7 = '8888-8888-8888-8888';  //all digits are the same.
+
+
+//remove dashes from credit card number.
+const removeDash = function (cardNum) {
+  remDash = cardNum.split('-').join('');
+  return remDash;
+}
+
+//check the sum of the card numbers.
+const checkSum = function (cardNum) {
+  let sum = 0;
+  for (let i = 0; i < cardNum.length; i++){
+    sum += parseInt(cardNum[i]);
+  }
+  return sum;
+}
+
+const validateCreditCard = function (creditCardNum) {
+  const formattedNum = removeDash(creditCardNum);
+  //check card number is 16 digits and check card number is all numbers and check final digit is even.
+  if (formattedNum.length !== 16 || isNaN(formattedNum) || formattedNum[formattedNum.length - 1] % 2 !== 0){
+    console.log(false);
+  } else if (checkSum(formattedNum) <= 16) {
+    console.log(false);
+  //check if atleast 2 digits are different.
+} else if  (formattedNum === '0000000000000000' ||
```

Comment:
Hey Daniel, I think this approach for checking if all digits are the same is great. You asked how you could take up less lines, one way would be to use a [set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set). Sets are like arrays, but can only contain one of each value, e.g. converting [1,2,2,3] to a set would result is 1,2,3. You can convert the array to a set, then check the length of the set, if it is 1 then all values must be the same. This isn't something we'll use often though. Congrats on getting through week 1!